### PR TITLE
helm: update tolerations and nodeSelectors

### PR DESCRIPTION
**Pull request by jackfrancis**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

This PR updates node tolerations and nodeSelectors in the helm chart so that the implementation works with 1.24.

Also removed an unnecessary toleration for the Windows cloud-node-manager deployment (Windows control plane nodes are not supported AFAIK).

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: helm/cloud-provider-azure/templates/cloud-provider-azure.yaml [hunk 1]
```diff
@@ -430,10 +438,6 @@ spec:
       tolerations:
       - key: CriticalAddonsOnly
         operator: Exists
-      - key: node-role.kubernetes.io/master
```
Comment (jackfrancis) [inline helm/cloud-provider-azure/templates/cloud-provider-azure.yaml:433]: @marosset @jsturtevant can you confirm that this control plane toleration is not needed for the Windows cloud-node-manager pod (i.e., there is no such thing as a "Windows control plane node" in the k8s ecosystem)

Comment (jsturtevant) [reply helm/cloud-provider-azure/templates/cloud-provider-azure.yaml:433]: That's correct, the windows `nodeSelector` above on line 437 is enough to land these on windows.

**Comment by jackfrancis** [comment]
@lzhecheng these updates are needed to enable functionality on 1.24 clusters

**Pull request merged**